include/cantera/base/Solution.h

```diff
@@ -20,16 +20,38 @@ class Solution : public std::enable_shared_from_this<Solution>
 {
 private:
     Solution();
+    Solution(const std::string& infile, std::string name, std::string transport,
+             std::vector<shared_ptr<Solution> > adjacent);
 
 public:
     ~Solution() {}
     Solution(const Solution&) = delete;
     Solution& operator=(const Solution&) = delete;
 
+    //! Create an empty Solution object
     static shared_ptr<Solution> create() {
         return shared_ptr<Solution>( new Solution );
     }
 
+    //! Create and initialize a Solution object from an input file
+    /*!
+     * This constructor wraps newPhase(), newKinetics() and
+     * newTransportMgr() routines for initialization.
+     *
+     * @param infile name of the input file
+     * @param name   name of the phase in the file.
+     *               If this is blank, the first phase in the file is used.
+     * @param transport name of the transport model.
+     * @param adjacent vector containing adjacent solution objects.
+     * @returns an initialized Solution object.
+     */
+    static shared_ptr<Solution> create(const std::string& infile,
+                                       std::string name="",
+                                       std::string transport="",
+                                       std::vector<shared_ptr<Solution> > adjacent={}) {
+        return shared_ptr<Solution>( new Solution(infile, name, transport, adjacent) );
+    }
```

Comment:
Not sure. I mostly introduced the `newSolution` for readability and consistency with other code. The function is inline, so the compiler knows what to do, i.e. I prefer to use to use a dedicated constructor to do the actual processing.

Reply:
I agree that the `newSolution` function is easier to use than `Solution::create` (people seem to struggle with how to call static methods in C++) and is consistent with the other methods like `newPhase`.

I'm not concerned about the performance of any particular implementation, but about simplicity. I don't the advantage to doing the initialization in the constructor, rather than the `newSolution` method. If we consolidate these methods, then there's never any question about whether to use `Solution::create` or `newSolution`, or why there's this constructor that can't be called.

Reply:
Done.